apps/meteor/app/api/server/lib/users.ts

```diff
@@ -111,6 +112,10 @@ export function getNonEmptyQuery(query: Query): typeof _defaultQuery | (typeof _
 		],
 	};
 
+	if (!canSeeAllUserInfo) {
+		defaultQuery.$or = defaultQuery.$or.filter((or) => !or['emails.address']);
+	}
+
```

Comment:
Can you invert the logic here please? easier to read `if hasPermission then push elem to array` than `if not hasPermission then create new array by filtering out xyz field` kinda over-wording this but I hope you're getting the point 🙈

Reply:
updated!